### ISSUE TITLE
please use CodeAction for related code generation

### ISSUE THREAD

**Issue by sslime336** [golang/vscode-go#2630]
**Is your feature request related to a problem? Please describe.**

Umm... no? The snippet like `meth` is helpful but I think below looks better.

**Describe the solution you'd like**

Like rust-analyzer and Goland, we can simply press `Ctrl + .` or `Alt + Enter` to generate a method of our own data structure.

RA(rust-analyzer):

![ra_codeaction_gen](https://user-images.githubusercontent.com/97684920/215334589-eb0c6c87-2076-476c-ba61-a0a4c028f992.gif)

Goland:

![goland_codeaction_gen](https://user-images.githubusercontent.com/97684920/215334598-9a22db2c-4d4a-4536-ad49-62df179a3a25.gif)

**Describe alternatives you've considered**
None.

**Additional context**

Actually I've written some very simple code to achieve this for fun, but my code is in poor quality and even the code actions can be waken up before the diagnosis! (and I am very new to vscode extension dev!)

Look good(?

![myex_codeaction_gen](https://user-images.githubusercontent.com/97684920/215334618-e447b1da-19f7-4831-9b8c-8d9f3e0e13d0.gif)

but below it looks like a fool :P

![myex_fail_codeaction_gen](https://user-images.githubusercontent.com/97684920/215334633-c293eb1d-8b90-4f66-a53f-50126b174b35.gif)

I've only looked into the repo for few days, so I'm not pretty sure how to achieve this properly.

Due to the project structure I've already learned, maybe there would be a internal break?

(noob i am~)

**Comment by hyangah** [comment]
Thanks for filing feature requests @sslime336 
I see three different code actions

* Rustlang: generating impl based on rust struct. A similar one can be a code action on Go interface type, that generates a struct type. 

* Goland's impl: This is similar to what the extension already offers https://github.com/golang/go/issues/37537  or https://github.com/golang/vscode-go/issues/2364#issuecomment-1192823013 Work to improve accessibility is necessary. Overcoming LSP limitation is currently the biggest challenge but we can also specialize for VS Code.

* Your demo code: Code action on a struct type that generates a new method.

**Comment by sslime336** [comment]
yeah, using CodeActions, I mean, just press Ctrl plus "." or Alt + Enter like Goland and select the options we needed may be more convenient :)

Actually I haven't thought about generateing a struct from an interface as you mentioned :P,  that's a possible option (but I don't know how to find out where to generate the struct that will impl the selected interface, especially when the interface is in a read-only file, we need to figure out the correct place)

Currently, I have these ideas where the CodeAction can be used:

1. Generate methods from an exist user defined type, just like my demo.
2. Implement an interface for an exist user defined type (firstly it will invoke a quick pick like in the issues you mentioned).
3. Implement an interface for an user defined type, but the CodeAction will be invoked from an local or global variable. 
4. As you first pointed, "generates a struct type" from an exist interface.

I'm not familiar with gopls so currently I could only think the ways to impl these features using TypeScript. But I'd like to learn more and mk a contribution as I depend on vscode-go seriously :-) (ah, this is a face with nose!)
